internal/controller/costmanagementmetricsconfig_controller_test.go

```diff
@@ -1448,4 +1453,125 @@ var _ = Describe("MetricsConfigController - CRD Handling", Ordered, func() {
 
 		})
 	})
+
+	Context("Source validation for uploads", func() {
+		BeforeEach(func() {
+			checkPVC = true
+		})
+
+		It("should skip upload when no valid source exists", func() {
+			Expect(setup()).Should(Succeed())
+			instCopy.Spec.Source.SourceName = ""
+			createObject(ctx, instCopy)
+
+			fetched := &metricscfgv1beta1.MetricsConfig{}
+
+			Eventually(func() bool {
+				_ = k8sClient.Get(ctx, types.NamespacedName{Name: instCopy.Name, Namespace: namespace}, fetched)
+				return fetched.Status.Authentication.AuthenticationCredentialsFound != nil
+			}, timeout, interval).Should(BeTrue())
+
+			// Without source name, SourceDefined should remain nil and upload should be blocked
+			Eventually(func() string {
+				_ = k8sClient.Get(ctx, types.NamespacedName{Name: instCopy.Name, Namespace: namespace}, fetched)
+				return fetched.Status.Upload.UploadError
+			}, timeout, interval).Should(ContainSubstring("Reports are being stored"))
+
+			Expect(fetched.Status.APIURL).To(Equal(defaultAPIURL))
+			Expect(fetched.Status.Upload.LastSuccessfulUploadTime.IsZero()).To(BeTrue())
+		})
+
+		It("should store reports when source validation fails", func() {
+			Expect(setup()).Should(Succeed())
+			instCopy.Spec.Source.SourceName = sourceName
+			instCopy.Spec.Source.CreateSource = &trueValue
+			createObject(ctx, instCopy)
+
+			fetched := &metricscfgv1beta1.MetricsConfig{}
+
+			Eventually(func() bool {
+				_ = k8sClient.Get(ctx, types.NamespacedName{Name: instCopy.Name, Namespace: namespace}, fetched)
+				return fetched.Status.Authentication.AuthenticationCredentialsFound != nil
+			}, timeout, interval).Should(BeTrue())
+
+			// Should get source validation error and block upload
```

Comment:
Again, this test is not configured to throw validation errors, so this statement is not correct.